lib/hotel.rb

```diff
@@ -0,0 +1,101 @@
+require_relative "room"
+# require_relative "customer"
+require_relative "duration"
+require_relative "reservation"
+require_relative "date_range"
+
+module HotelBooking
+  class Hotel
+    TOTAL_ROOMS = 20
+    attr_reader :id, :rooms, :reservations, :vacancy
+    attr_accessor :customers
+
+    def initialize(id:, name:)
+      @id = id
+      @name = name
+      @rooms = []
+      @reservations = []
+      @customers = []
+      @vacancy = true
+
+      make_room
+    end
+
+    def make_room()
+      TOTAL_ROOMS.times do |index|
+        room = Room.new(id: index + 1, number: index + 1, availability: true, reservation: nil)
+        @rooms << room
+      end
+    end
+
+    def get_total_rooms()
+      return @rooms.length
+    end
+
+    def add_customer(customer)
+      # customer = Customer.new(id: 1, name: "one")
+      @customers << customer
+    end
+
+    def find_customer(id)
+      customer = @customers.find { |customer| customer.id == id }
+      return customer
+    end
+
+    #list of rooms that are not reserverd
+    def get_availbale_rooms(date)
+      available_rooms = @rooms.find_all { |room| room.availability == true }
+      # rooms_date_range = available_rooms.find_all { |room| (date >= room.reservation.duration.start_date && date <= room.reservation.duration.end_date) }
+      return available_rooms
```

Comment:
This method doesn't use the `date` parameter. Why not? What does it mean for a room to have `availability`? Isn't the requirement to find rooms that are available for a specific date? What does it mean when a room is "available"?